soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -60,9 +64,12 @@ static int nordicsemi_nrf53_init(const struct device *arg)
 #endif
 
 #if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
```

Comment:
You could name the Kconfig as NRF5340_CPUAPP_ENABLE_LFXO and remove the other macros from the preprocessor expression

Reply:
Yes, I could. But I'm not sure if that will be better. Currently, you can clearly see in the source code when this part will be executed. After I move the dependencies to Kconfig, this will be a bit obscured. And if `SOC_ENABLE_LFXO` becomes dependent on `!TRUSTED_EXECUTION_NONSECURE`, it won't be possible to use a common configuration that changes this option in builds for secure and non-secure boards (it won't be accessible in the latter case).

Reply:
>And if SOC_ENABLE_LFXO becomes dependent on !TRUSTED_EXECUTION_NONSECURE, it won't be possible to use a common configuration that changes this option in builds for secure and non-secure boards (it won't be accessible in the latter case)

It should not be accessible, right? Cause only Secure  code has access to it.
It could be accessible in non-TrustZone SoCs, if that's what you mean...

Reply:
I'm leaning towards adding filter in Kconfig, since this option should not be visible in menuconfig for nonsecure app.

Other options like `CONFIG_SOC_DCDC_NRF53X_APP` does not have string in kconfig.

Reply:
> Other options like `CONFIG_SOC_DCDC_NRF53X_APP` does not have string in kconfig.

`SOC_ENABLE_LFXO` intentionally has a prompt, as it needs to be possible for users to change it directly in their project configurations.

Reply:
@ioannisg @nordic-krch Regarding dependency on `!TRUSTED_EXECUTION_NONSECURE`. I was thinking about an application (let's say some sample) with the following in its `prj.conf`:
```
CONFIG_SOC_ENABLE_LFXO=n
```
With such dependency, it would not be possible to build it for both `_cpuapp` and `_cpuappns` targets. But maybe I'm overthinking it.

Reply:
> it would not be possible to use it for both _cpuapp and _cpuappns targets. But maybe I'm overthinking it.

Question: is CONFIG_SOC_ENABLE_LFXO an option that will only be defined for nRF platforms or it should be generic enough to be used in any SoC or Board?

Reply:
IMO it's an nRF specific option.

Reply:
Then, in my opinion such option should
- have an NRFxxxx in-fix, to make it clear it is an nRF specific option, and
- not be present in a generic Zephyr (cross-platform) sample;

If a user that develops on nRF platform wants to set this to =n, they can do it in a sample Kconfig (and then they can have the TEE dependency).

But I am not really an application developer :)